File: ios/SwiftKotlination/SwiftKotlinationTests/Sources/Features/Story/Model/Mocks/StoryManagerMock.swift
import XCTest
@testable import SwiftKotlination

struct StoryManagerMock {
    var result: Result<Story, Error>

    init(result: Result<Story, Error> = .failure(NetworkError.invalidResponse)) {
        self.result = result
    }
}

extension StoryManagerMock: StoryManagerProtocol {
    func story(_ observer: @escaping Observer<Story>) {
        observer(result)
    }
}

File: android/SwiftKotlination/app/src/test/kotlin/fr/jhandguy/swiftkotlination/features/story/model/mocks/StoryManagerMock.kt
package fr.jhandguy.swiftkotlination.features.story.model.mocks

import fr.jhandguy.swiftkotlination.features.story.model.Story
import fr.jhandguy.swiftkotlination.features.story.model.StoryManagerInterface
import fr.jhandguy.swiftkotlination.observer.Observer
import fr.jhandguy.swiftkotlination.observer.Result

class StoryManagerMock(val result: Result<Story>) : StoryManagerInterface {
    override suspend fun story(observer: Observer<Story>) = observer(result)
}
